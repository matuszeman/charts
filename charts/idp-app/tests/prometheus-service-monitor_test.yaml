suite: prometheus-service-monitor
templates:
  - prometheus-service-monitor.yaml
values:
  - values/minimum.yaml
tests:
  - it: when prometheus serviceMonitor is enabled but not service
    set:
      prometheus:
        serviceMonitor:
          enabled: true
    asserts:
      - failedTemplate:
          errorMessage: "prometheus.serviceMonitor: service must be enabled"
  - it: snapshot
    values:
      - values/service.yaml
      - values/deployment.yaml
      - values/prometheus.yaml
    asserts:
      - matchSnapshot: { }
  - it: when multi deployment
    values:
      - values/service.yaml
      - values/deployment.yaml
      - values/prometheus.yaml
      - values/multi.yaml
    asserts:
      - matchSnapshot: { }
      - equal:
          path: metadata.name
          value: RELEASE-NAME-one
        documentIndex: 0
      - equal:
          path: metadata.name
          value: RELEASE-NAME-two
        documentIndex: 1

