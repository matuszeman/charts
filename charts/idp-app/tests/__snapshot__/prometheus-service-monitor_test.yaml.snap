snapshot:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME
    spec:
      endpoints:
        - path: /metrics
          targetPort: prometheus-exporter
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME
          app.kubernetes.io/name: app-name
when multi deployment:
  1: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-one
    spec:
      endpoints:
        - path: /metrics
          targetPort: prometheus-exporter
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME-one
          app.kubernetes.io/name: app-name
  2: |
    apiVersion: monitoring.coreos.com/v1
    kind: ServiceMonitor
    metadata:
      labels:
        app.kubernetes.io/instance: RELEASE-NAME
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: app-name
        app.kubernetes.io/version: 0.1.0
      name: RELEASE-NAME-two
    spec:
      endpoints:
        - path: /metrics
          targetPort: prometheus-exporter
      namespaceSelector:
        matchNames:
          - NAMESPACE
      selector:
        matchLabels:
          app.kubernetes.io/instance: RELEASE-NAME-two
          app.kubernetes.io/name: app-name
